File: src/lib/types/gammes.ts
// Types gammes de maintenance — découplage gammes/équipements

// ── Domaines et familles gammes (hiérarchie indépendante des équipements) ──

export interface DomaineGamme {
  id_domaine_gamme: number;
  nom_domaine: string;
  description: string | null;
  id_image: number | null;
}

export interface FamilleGamme {
  id_famille_gamme: number;
  nom_famille: string;
  description: string | null;
  id_domaine_gamme: number;
  id_image: number | null;
}

export interface DomaineGammeListItem {
  id_domaine_gamme: number;
  nom_domaine: string;
  description: string | null;
  id_image: number | null;
  nb_familles: number;
  nb_gammes_inactives: number;
  nb_gammes_total: number;
  nb_ot_en_retard: number;
  nb_ot_reouvert: number;
  nb_ot_en_cours: number;
  nb_ot_sans_ot: number;
  prochaine_date: string | null;
  jours_periodicite_min: number | null;
}

export interface FamilleGammeListItem {
  id_famille_gamme: number;
  nom_famille: string;
  description: string | null;
  id_image: number | null;
  nb_gammes: number;
  nb_gammes_inactives: number;
  nb_ot_en_retard: number;
  nb_ot_reouvert: number;
  nb_ot_en_cours: number;
  nb_ot_sans_ot: number;
  prochaine_date: string | null;
  jours_periodicite_min: number | null;
}

// ── Gammes ──

export interface Gamme {
  id_gamme: number;
  nom_gamme: string;
  description: string | null;
  est_reglementaire: number;
  id_periodicite: number;
  id_famille_gamme: number;
  id_prestataire: number;
  id_image: number | null;
  id_batiment_calc: number | null;
  id_niveau_calc: number | null;
  id_local_calc: number | null;
  nom_localisation_calc: string | null;
  date_creation: string | null;
  date_modification: string | null;
  est_active: number;
}

export interface GammeListItem {
  id_gamme: number;
  nom_gamme: string;
  est_reglementaire: number;
  est_active: number;
  nom_famille: string;
  libelle_periodicite: string;
  nom_prestataire: string;
  description: string | null;
  id_image: number | null;
  nb_documents: number;
  nb_ot_total: number;
  nb_ot_en_retard: number;
  nb_ot_reouvert: number;
  nb_ot_en_cours: number;
  prochaine_date: string | null;
  jours_periodicite: number;
}

export interface Operation {
  id_operation: number;
  nom_operation: string;
  description: string | null;
  id_type_operation: number;
  id_gamme: number;
  seuil_minimum: number | null;
  seuil_maximum: number | null;
  id_unite: number | null;
}

export interface ModeleOperation {
  id_modele_operation: number;
  nom_modele: string;
  description: string | null;
  id_image: number | null;
  date_creation: string | null;
}

export interface ModeleOperationItem {
  id_modele_operation_item: number;
  nom_operation: string;
  description: string | null;
  id_type_operation: number;
  id_modele_operation: number;
  seuil_minimum: number | null;
  seuil_maximum: number | null;
  id_unite: number | null;
}

File: src-tauri/src/models/gammes.rs
use serde::{Deserialize, Serialize};

// ── Domaines gammes (classification indépendante des équipements) ──

#[derive(Debug, Serialize, Deserialize)]
pub struct DomaineGamme {
    pub id_domaine_gamme: i64,
    pub nom_domaine: String,
    pub description: Option<String>,
    pub id_image: Option<i64>,
}

#[derive(Debug, Deserialize)]
pub struct DomaineGammeInput {
    pub nom_domaine: String,
    pub description: Option<String>,
    pub id_image: Option<i64>,
}

// ── Familles gammes ──

#[derive(Debug, Serialize, Deserialize)]
pub struct FamilleGamme {
    pub id_famille_gamme: i64,
    pub nom_famille: String,
    pub description: Option<String>,
    pub id_domaine_gamme: i64,
    pub id_image: Option<i64>,
}

#[derive(Debug, Deserialize)]
pub struct FamilleGammeInput {
    pub nom_famille: String,
    pub description: Option<String>,
    pub id_domaine_gamme: i64,
    pub id_image: Option<i64>,
}

/// Version liste enrichie pour affichage en cartes (domaines)
#[derive(Debug, Serialize, Deserialize)]
pub struct DomaineGammeListItem {
    pub id_domaine_gamme: i64,
    pub nom_domaine: String,
    pub description: Option<String>,
    pub id_image: Option<i64>,
    pub nb_familles: i64,
    pub nb_gammes_inactives: i64,
    pub nb_gammes_total: i64,
    pub nb_ot_en_retard: i64,
    pub nb_ot_reouvert: i64,
    pub nb_ot_en_cours: i64,
    pub nb_ot_sans_ot: i64,
    pub prochaine_date: Option<String>,
    pub jours_periodicite_min: Option<i64>,
}

/// Version liste enrichie pour affichage en cartes (familles)
#[derive(Debug, Serialize, Deserialize)]
pub struct FamilleGammeListItem {
    pub id_famille_gamme: i64,
    pub nom_famille: String,
    pub description: Option<String>,
    pub id_image: Option<i64>,
    pub nb_gammes: i64,
    pub nb_gammes_inactives: i64,
    pub nb_ot_en_retard: i64,
    pub nb_ot_reouvert: i64,
    pub nb_ot_en_cours: i64,
    pub nb_ot_sans_ot: i64,
    pub prochaine_date: Option<String>,
    pub jours_periodicite_min: Option<i64>,
}

// ── Gammes (procédures de maintenance) ──

#[derive(Debug, Serialize, Deserialize)]
pub struct Gamme {
    pub id_gamme: i64,
    pub nom_gamme: String,
    pub description: Option<String>,
    pub est_reglementaire: i64,
    pub id_periodicite: i64,
    pub id_famille_gamme: i64,
    pub id_prestataire: i64,
    pub id_image: Option<i64>,
    pub id_batiment_calc: Option<i64>,
    pub id_niveau_calc: Option<i64>,
    pub id_local_calc: Option<i64>,
    pub nom_localisation_calc: Option<String>,
    pub date_creation: Option<String>,
    pub date_modification: Option<String>,
    pub est_active: i64,
}

#[derive(Debug, Deserialize)]
pub struct GammeInput {
    pub nom_gamme: String,
    pub description: Option<String>,
    pub est_reglementaire: i64,
    pub id_periodicite: i64,
    pub id_famille_gamme: i64,
    pub id_prestataire: i64,
    pub id_image: Option<i64>,
}

/// Version liste avec noms résolus
#[derive(Debug, Serialize, Deserialize)]
pub struct GammeListItem {
    pub id_gamme: i64,
    pub nom_gamme: String,
    pub est_reglementaire: i64,
    pub est_active: i64,
    pub nom_famille: String,
    pub libelle_periodicite: String,
    pub nom_prestataire: String,
    pub description: Option<String>,
    pub id_image: Option<i64>,
    pub nb_documents: i64,
    pub nb_ot_total: i64,
    pub nb_ot_en_retard: i64,
    pub nb_ot_reouvert: i64,
    pub nb_ot_en_cours: i64,
    pub prochaine_date: Option<String>,
    pub jours_periodicite: i64,
}

// ── Opérations spécifiques d'une gamme ──

#[derive(Debug, Serialize, Deserialize)]
pub struct Operation {
    pub id_operation: i64,
    pub nom_operation: String,
    pub description: Option<String>,
    pub id_type_operation: i64,
    pub id_gamme: i64,
    pub seuil_minimum: Option<f64>,
    pub seuil_maximum: Option<f64>,
    pub id_unite: Option<i64>,
}

#[derive(Debug, Deserialize)]
pub struct OperationInput {
    pub nom_operation: String,
    pub description: Option<String>,
    pub id_type_operation: i64,
    pub id_gamme: i64,
    pub seuil_minimum: Option<f64>,
    pub seuil_maximum: Option<f64>,
    pub id_unite: Option<i64>,
}
